rxgroups-android/src/main/java/com/airbnb/rxgroups/LifecycleResubscription.java

```diff
@@ -62,21 +79,47 @@ private boolean shouldCheckClass(@Nullable Class<?> clazz) {
     return !qualifiedName.startsWith("android.") && !qualifiedName.startsWith("java.");
   }
 
-  private Observable<ObserverInfo> observerInfoList(Object target, Field field) {
+  private Observable<ObserverInfo> observerInfoList(final Object target, Field field) {
     final Observer<?> observer;
     try {
       observer = (Observer<?>) field.get(target);
     } catch (IllegalAccessException e) {
       throw new RuntimeException(
-          String.format("Error accessing observer %s. Make sure it's no private.", field), e);
+          String.format("Error accessing observer %s. Make sure it's not private.", field), e);
     }
 
-    return Observable.from(field.getAnnotation(AutoResubscribe.class).value())
-        .map(new Func1<String, ObserverInfo>() {
-          @Override public ObserverInfo call(String tag) {
-            return new ObserverInfo(tag, observer);
+    return Observable.just(field).flatMap(new Func1<Field, Observable<ObserverInfo>>() {
+      @Override public Observable<ObserverInfo> call(Field field) {
+        //noinspection TryWithIdenticalCatches
+        try {
+          Class<?> fieldClass = observer.getClass();
+          Method tagMethod = fieldClass.getDeclaredMethod("resubscriptionTag");
+          Object tag = tagMethod.invoke(observer);
+          Class<?> rawParameterType = Utils.getRawType(tag.getClass());
+          Func1<Object, ObserverInfo> collectionMapper = new Func1<Object, ObserverInfo>() {
+            @Override public ObserverInfo call(Object s) {
+              return new ObserverInfo(s.toString(), observer);
+            }
+          };
+          //noinspection ConstantConditions
+          if (Iterable.class.isAssignableFrom(rawParameterType)) {
+            return Observable.from((Iterable<?>) tag).map(collectionMapper);
           }
-        });
+          if (rawParameterType.isArray()) {
+            return Observable.from(Utils.boxIfPrimitiveArray(tag)).map(collectionMapper);
+          }
+          return Observable.just(new ObserverInfo(tag.toString(), observer));
+        } catch (NoSuchMethodException e) {
+          throw new RuntimeException("Please define a method named 'resubscriptionTag()'", e);
+        } catch (InvocationTargetException e) {
+          throw new RuntimeException(
+              "Error accessing method 'resubscriptionTag()'. Make sure it's not private", e);
```

Comment:
`"Error accessing method 'resubscriptionTag()'. Make sure it's not private"` is duplicated a lot. we could make our own exception